controllers/object.js

```diff
@@ -63,6 +63,58 @@ const uploadVideo = function(objectID, videoID, reqForForm, callback) {
     }
 };
 
+function uploadMediaFile(objectID, req, callback) {
+    console.log('received media file for', objectID);
+
+    let object = utilities.getObject(objects, objectID);
+    if (!object) {
+        callback(404, 'object ' + objectID + ' not found');
+        return;
+    }
+
+    var mediaDir = objectsPath + '/' + object.name + '/' + identityFolderName + '/mediaFiles';
+    if (!fs.existsSync(mediaDir)) {
+        fs.mkdirSync(mediaDir);
+    }
+
+    var form = new formidable.IncomingForm({
+        uploadDir: mediaDir,
+        keepExtensions: true
+        // accept: 'image/jpeg' // TODO: specify which types of images/videos it accepts?
+    });
+
+    console.log('created form');
+
+    form.on('error', function (err) {
+        callback(500, err);
+    });
+
+    let mediaUuid = utilities.uuidTime();
+
+    var rawFilepath = form.uploadDir + '/' + mediaUuid + '.jpg';
+
+    if (fs.existsSync(rawFilepath)) {
+        console.log('deleted old raw file');
+        fs.unlinkSync(rawFilepath);
+    }
+
+    form.on('fileBegin', function (name, file) {
+        console.log('fileBegin loading', name, file);
+
+        if (file.type.match('video.*')) {
+            rawFilepath = rawFilepath.replace('.jpg', '.mov');
```

Comment:
Is there a less fragile way to do this? Or is this working around a specific jpg <-> mov bug?

Reply:
Yeah I can do this in a less fragile way